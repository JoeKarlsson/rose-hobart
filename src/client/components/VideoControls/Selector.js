import React, { Component } from 'react';
import PropTypes from 'prop-types';

const styles = {
	root: {
		display: 'flex',
		flexDirection: 'row',
	},
	select: {
		flex: 1,
	},
};

class Selector extends Component {

	onChange(e) {
		e.preventdefault();
		this.props.onChange(e.target.value);
	}
	random() {
		const { onChange, choices } = this.props;
		onChange(choices[Math.floor(Math.random() * choices.length)].key);
	}
	prev() {
		const { onChange, choices } = this.props;
		const keys = choices.map(c => c.key);
		const index = keys.indexOf(this.props.value);
		const lastIndex = keys.length - 1;
		onChange(keys[(index + lastIndex) % keys.length]);
	}

	next() {
		const { onChange, choices } = this.props;
		const keys = choices.map(c => c.key);
		const index = keys.indexOf(this.props.value);
		onChange(keys[(index + 1) % keys.length]);
	}

	render() {
		const { value, choices } = this.props;
		return (
			<div style={styles.root}>
				<button onClick={this.prev}>←</button>
				<select style={styles.select} value={value} onChange={this.onChange}>
					{choices.map(({ key, label }) =>
						<option key={key} value={key}>{label}</option>)}
				</select>
				<button onClick={this.next}>→</button>
			</div>
		);
	}
}

Selector.propTypes = {
	choices: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
	onChange: PropTypes.func.isRequired,
};

Selector.defaultProps = {};

export default Selector;
