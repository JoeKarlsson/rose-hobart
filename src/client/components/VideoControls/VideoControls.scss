.VideoControls {

	.twitter {
		display: block;
		background: #3a3a3a;
		border-radius: 5px;
		font: bold italic 15px "Helvetica" italic;
		text-align: center;
		margin: 10px 0 0;
		padding: 15px 0;
		color: #5a5a5a;
		text-decoration: none;
		transition: 0.3s ease;
	}

	.twitter:hover {
		background: #414141;
		color: #4099ff;
	}
}
