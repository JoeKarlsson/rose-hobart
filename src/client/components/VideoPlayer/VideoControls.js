import React from 'react';
import {
	Player,
	ControlBar,
	ReplayControl,
	ForwardControl,
	CurrentTimeDisplay,
	TimeDivider,
	LoadingSpinner,
	PlaybackRateMenuButton,
	VolumeMenuButton,
} from 'video-react';
import 'video-react/dist/video-react.css';
import './VideoPlayer.scss';
import poster from '../../assets/images/poster.png';
import video from '../../assets/video/video.mp4';

export default () => {
	return (
		<div className="video_player">
			<Player
				poster={poster}
			>
				<source src={video} />
				<LoadingSpinner />

				<ControlBar>
					<ReplayControl seconds={10} order={1.1} />
					<ForwardControl seconds={30} order={1.2} />
					<CurrentTimeDisplay order={4.1} />
					<TimeDivider order={4.2} />
					<PlaybackRateMenuButton
						rates={[5, 2, 1, 0.5, 0.1]}
						order={7.1}
					/>
					<VolumeMenuButton disabled />
				</ControlBar>
			</Player>
		</div>
	);
};
