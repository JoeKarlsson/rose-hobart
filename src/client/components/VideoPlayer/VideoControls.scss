.VideoControls {

	aside {
		width: 360px;
		margin: 0 0 15px 15px;
	}

	aside div {
		padding: 10px;
		border-bottom: 1px solid #888;
		border-top: 1px solid #bbb;
		transition: background 0.3s ease;
		background: #aaa;
	}

	aside div:first-child {
		padding-top: 20px;
	}

	aside div:hover {
		background: #bbb;
	}

	label {
		display: block;
		font: 13px "Helvetica", sans-serif;
		color: #222;
	}

	strong {
		font-weight: bold;
		padding: 0 3px;
		user-select: none;
	}

	.twitter {
		display: block;
		background: #3a3a3a;
		border-radius: 5px;
		font: bold italic 15px "Helvetica" italic;
		text-align: center;
		margin: 10px 0 0;
		padding: 15px 0;
		color: #5a5a5a;
		text-decoration: none;
		transition: 0.3s ease;
	}

	.twitter:hover {
		background: #414141;
		color: #4099ff;
	}
}
