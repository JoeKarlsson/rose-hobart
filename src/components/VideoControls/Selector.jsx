import React, { Component } from 'react';
import PropTypes from 'prop-types';

const styles = {
	root: {
		display: 'flex',
		flexDirection: 'row',
	},
	select: {
		flex: 1,
	},
};

class Selector extends Component {
	constructor() {
		super();
		this.onChange = this.onChange.bind(this);
		this.prev = this.prev.bind(this);
		this.next = this.next.bind(this);
		this.random = this.random.bind(this);
	}

	onChange(e) {
		const { onChange } = this.props;
		onChange(e.target.value);
	}

	random() {
		const { onChange, choices } = this.props;
		onChange(choices[Math.floor(Math.random() * choices.length)].key);
	}

	prev(e) {
		e.preventDefault();
		const { onChange, choices, currColor } = this.props;
		const keys = choices.map(c => c.key);
		const index = keys.indexOf(currColor);
		const lastIndex = keys.length - 1;
		onChange(keys[(index + lastIndex) % keys.length]);
	}

	next(e) {
		e.preventDefault();
		const { onChange, choices, currColor } = this.props;
		const keys = choices.map(c => c.key);
		const index = keys.indexOf(currColor);
		onChange(keys[(index + 1) % keys.length]);
	}

	render() {
		const { choices, currColor } = this.props;

		return (
			<div style={styles.root}>
				<button onClick={this.prev}>←</button>
				<select style={styles.select} value={currColor} onChange={this.onChange}>
					{choices.map(({ key, label }) =>
						<option key={key} value={key}>{label}</option>)}
				</select>
				<button onClick={this.next}>→</button>
			</div>
		);
	}
}

Selector.propTypes = {
	currColor: PropTypes.string.isRequired, // eslint-disable-line react/forbid-prop-types
	choices: PropTypes.array.isRequired, // eslint-disable-line react/forbid-prop-types
	onChange: PropTypes.func.isRequired,
};

Selector.defaultProps = {};

export default Selector;
