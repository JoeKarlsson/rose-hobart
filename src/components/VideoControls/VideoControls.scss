// Dark Theater Video Controls - Subtle Dark Theme
.VideoControls,
.VideoControls *,
.VideoControls .row,
.VideoControls .col,
.VideoControls form,
.VideoControls div {
	background: transparent !important;
	color: #cccccc !important;
}

// Target the main container div specifically
.VideoControls {
	padding: 20px !important;
	background: #000000 !important;
	border: 1px solid #333333 !important;
	border-radius: 8px !important;
	margin: 20px 0 !important;
	box-shadow: 
		0 2px 8px rgba(0,0,0,0.3),
		inset 0 1px 0 rgba(255,255,255,0.05) !important;
	color: #cccccc !important;
	
	// Form styling
	form {
		background: transparent !important;
		color: #ffffff !important;
		
		.row {
			margin-bottom: 20px !important;
			background: transparent !important;
			
			&:last-child {
				margin-bottom: 0 !important;
			}
		}
	}
	
	// Override Materialize grid classes
	.row {
		background: #000000 !important;
		margin-left: 0 !important;
		margin-right: 0 !important;
		padding: 15px !important;
		border-radius: 6px !important;
		border: 1px solid #333333 !important;
	}
	
	.col {
		background: transparent !important;
	}
	
	.col.s12 {
		background: transparent !important;
	}
	
	// Target the specific div structure
	> .row {
		background: #000000 !important;
		border: 1px solid #333333 !important;
		border-radius: 6px !important;
		padding: 15px !important;
		margin-bottom: 15px !important;
	}
	
	// Labels - Subtle text
	label {
		color: #cccccc !important;
		font-family: 'Oswald', sans-serif;
		font-weight: 500;
		text-transform: uppercase;
		letter-spacing: 0.05em;
		text-shadow: 1px 1px 2px rgba(0,0,0,0.8);
		margin-bottom: 10px;
		display: block;
		
		strong {
			color: #ffffff;
			text-shadow: 0 0 5px rgba(255, 255, 255, 0.2);
		}
	}
	
	// Control buttons container
	.row {
		display: flex;
		flex-wrap: wrap;
		gap: 15px;
		align-items: center;
		margin-bottom: 15px;
		
		&:last-child {
			margin-bottom: 0;
		}
	}
	
	// Individual control sections
	> form > .row > div {
		background: rgba(0,0,0,0.2);
		padding: 15px;
		border-radius: 6px;
		border: 1px solid #333333;
		flex: 1;
		min-width: 200px;
		
		&:hover {
			background: rgba(0,0,0,0.4);
			border-color: #555555;
		}
	}
	
	// Subtle dark buttons
	.btn {
		background: #333333 !important;
		color: #ffffff !important;
		border: 1px solid #555555 !important;
		border-radius: 6px !important;
		padding: 10px 20px !important;
		font-family: 'Oswald', sans-serif !important;
		font-weight: 500 !important;
		text-transform: uppercase !important;
		letter-spacing: 0.05em !important;
		box-shadow: 
			0 2px 4px rgba(0,0,0,0.3),
			inset 0 1px 0 rgba(255,255,255,0.1) !important;
		transition: all 0.3s ease !important;
		margin: 5px !important;
		
		&:hover {
			background: #444444 !important;
			box-shadow: 
				0 4px 8px rgba(0,0,0,0.4),
				0 0 10px rgba(255, 255, 255, 0.1),
				inset 0 1px 0 rgba(255,255,255,0.2) !important;
			transform: translateY(-1px) scale(1.02) !important;
		}
		
		&:active {
			transform: translateY(0) scale(1.01) !important;
		}
		
		.material-icons {
			color: #ffffff !important;
			font-size: 18px !important;
		}
	}
	
	// Subtle dark sliders
	input[type="range"] {
		-webkit-appearance: none;
		appearance: none;
		height: 8px;
		background: #333333;
		border-radius: 4px;
		outline: none;
		border: 1px solid #555555;
		box-shadow: inset 0 0 10px rgba(0,0,0,0.5);
		
		&::-webkit-slider-thumb {
			-webkit-appearance: none;
			appearance: none;
			width: 20px;
			height: 20px;
			background: #666666;
			border-radius: 50%;
			border: 2px solid #333333;
			box-shadow: 
				0 0 5px rgba(255, 255, 255, 0.1),
				0 2px 4px rgba(0,0,0,0.3);
			cursor: pointer;
			transition: all 0.3s ease;
			
			&:hover {
				background: #777777;
				box-shadow: 
					0 0 8px rgba(255, 255, 255, 0.2),
					0 4px 8px rgba(0,0,0,0.4);
				transform: scale(1.05);
			}
		}
		
		&::-moz-range-thumb {
			width: 20px;
			height: 20px;
			background: #666666;
			border-radius: 50%;
			border: 2px solid #333333;
			box-shadow: 
				0 0 5px rgba(255, 255, 255, 0.1),
				0 2px 4px rgba(0,0,0,0.3);
			cursor: pointer;
			transition: all 0.3s ease;
			
			&:hover {
				background: #777777;
				box-shadow: 
					0 0 8px rgba(255, 255, 255, 0.2),
					0 4px 8px rgba(0,0,0,0.4);
				transform: scale(1.05);
			}
		}
	}
	
	// Selector styling
	select {
		background: #000000 !important;
		color: #cccccc !important;
		border: 1px solid #555555 !important;
		border-radius: 6px !important;
		padding: 10px 15px !important;
		font-family: 'Oswald', sans-serif !important;
		font-weight: 500 !important;
		text-transform: uppercase !important;
		letter-spacing: 0.05em !important;
		box-shadow: 
			0 0 5px rgba(0, 0, 0, 0.3),
			inset 0 0 10px rgba(0,0,0,0.5) !important;
		
		&:focus {
			outline: none !important;
			box-shadow: 
				0 0 8px rgba(255, 255, 255, 0.1),
				inset 0 0 10px rgba(0,0,0,0.7) !important;
		}
		
		option {
			background: #000000 !important;
			color: #cccccc !important;
		}
	}
	
	// Override any Materialize input styling
	input, textarea, select {
		background: #000000 !important;
		color: #cccccc !important;
		border: 1px solid #555555 !important;
	}
	
	// Override any Materialize card or panel styling
	.card, .card-panel {
		background: transparent !important;
		color: #cccccc !important;
	}
	
	// Force dark theme on all child elements
	* {
		color: #cccccc !important;
	}
	
	// Override any white backgrounds
	div, span, p, label, strong {
		background: transparent !important;
		color: #cccccc !important;
	}
	
	// Force dark background on all divs within VideoControls
	div {
		background: #000000 !important;
		color: #cccccc !important;
	}
	
	// Override any Materialize white backgrounds
	.row, .col, .card, .card-panel, .input-field, .select-wrapper {
		background: #000000 !important;
		color: #cccccc !important;
	}
}

// Additional global override for VideoControls container
.VideoControls div[class*="row"],
.VideoControls div[class*="col"],
.VideoControls div[class*="card"] {
	background: #000000 !important;
	color: #cccccc !important;
}
