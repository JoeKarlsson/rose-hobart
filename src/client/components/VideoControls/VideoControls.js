/*
  eslint
	jsx-a11y/label-has-for: 0
*/

// https://codepen.io/bennettfeely/pen/kmhBI?depth=everything&order=popularity&page=9&q=video&show_forks=false

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StaticBlurMap from './StaticBlurMap/StaticBlurMap';
import FloatSlider from './FloatSlider/FloatSlider';
import VideoControlButton from './VideoControlButton/VideoControlButton';
import hash from '../../helper/hash';
import './VideoControls.scss';

// We must use a <Bus> if we don't want the <video> element to be duplicated
// per Blur pass.. Also since we can dynamically change the number of passes,
// it changes the tree level, (e.g. Blur1D>Blur1D>video becomes Blur1D>video)
// and React always destroys and recreates the instance during reconcialition.
class VideoControls extends Component {
	constructor() {
		super();
		this.state = {};

		this.controls = ['play', 'pause', 'mute'];
	}

	render() {
		const {
			factor,
			passes,
			contrast,
			saturation,
			brightness,
			onStaticBlurMapChange,
			handleClick,
			onChange,
		} = this.props;

		return (
			<div className="VideoControls row">
				<form className="col s12">
					<div className="row">
						{this.controls.map(control => (
							<VideoControlButton
								key={hash(control)}
								control={control}
								onItemClick={handleClick}
							/>
						))}
						<div>
							<label>
								<strong>Blur Factor</strong>
							</label>
							<FloatSlider
								title="factor"
								min={0}
								max={8}
								step={0.2}
								handleChange={onChange}
								value={factor}
							/>
						</div>
						<div>
							<label><strong>Blur Passes</strong></label>
							<FloatSlider
								title="passes"
								min={1}
								max={8}
								step={1}
								handleChange={onChange}
								value={passes}
							/>
						</div>
						<div>
							<label><strong>Saturation</strong></label>
							<FloatSlider
								title="saturation"
								min={0}
								max={2}
								step={0.05}
								handleChange={onChange}
								value={saturation}
							/>
						</div>
						<div>
							<label><strong>Brightness</strong></label>
							<FloatSlider
								title="brightness"
								min={0}
								max={2}
								step={0.05}
								handleChange={onChange}
								value={brightness}
							/>
						</div>
						<div>
							<label><strong>Contrast</strong></label>
							<FloatSlider
								title="contrast"
								min={0}
								max={2}
								step={0.05}
								handleChange={onChange}
								value={contrast}
							/>
						</div>

						<StaticBlurMap handleChange={onStaticBlurMapChange} />

					</div>
				</form>
			</div>
		);
	}
}

VideoControls.propTypes = {
	factor: PropTypes.number.isRequired,
	passes: PropTypes.number.isRequired,
	contrast: PropTypes.number.isRequired,
	saturation: PropTypes.number.isRequired,
	brightness: PropTypes.number.isRequired,
	onStaticBlurMapChange: PropTypes.func.isRequired,
	handleClick: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
};

VideoControls.defaultProps = {};

export default VideoControls;
