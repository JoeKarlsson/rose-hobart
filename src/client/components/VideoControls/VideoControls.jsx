/*
  eslint
	jsx-a11y/label-has-for: 0
*/

import React from 'react';
import PropTypes from 'prop-types';
import { NearestCopy } from 'gl-react';
import { Surface } from 'gl-react-dom';
import StaticBlurMap from './StaticBlurMap/StaticBlurMap.jsx';
import FloatSlider from './FloatSlider/FloatSlider';
import VideoControlButton from './VideoControlButton/VideoControlButton';
import Selector from './Selector';
import colorScales from '../VideoPlayer/helper/colorScales';
import hash from '../../helper/hash';
import './VideoControls.scss';

const ToolboxFooter = ({ color }) => (
	<Surface style={{ minWidth: '100%' }} width={400} height={20}>
		<NearestCopy>{colorScales[color]}</NearestCopy>
	</Surface>
);

ToolboxFooter.propTypes = {
	color: PropTypes.string.isRequired,
};

ToolboxFooter.defaultProps = {};

// We must use a <Bus> if we don't want the <video> element to be duplicated
// per Blur pass.. Also since we can dynamically change the number of passes,
// it changes the tree level, (e.g. Blur1D>Blur1D>video becomes Blur1D>video)
// and React always destroys and recreates the instance during reconcialition.
const VideoControls = ({
	color,
	factor,
	passes,
	contrast,
	saturation,
	brightness,
	onStaticBlurMapChange,
	handleColorChange,
	handleClick,
	onChange,
}) => {
	const controls = ['play', 'pause', 'mute', 'unmute', 'slow', 'fast', 'original'];

	const choices = Object.keys(colorScales).map((cs) => {
		return { key: cs, label: cs };
	});

	return (
		<div className="VideoControls row">
			<form className="col s12">
				<div className="row">
					<ToolboxFooter color={color} />

					<Selector choices={choices} onChange={handleColorChange} currColor={color} />
				</div>
				<div className="row">
					{controls.map(control => (
						<VideoControlButton
							key={hash(control)}
							control={control}
							onItemClick={handleClick}
						/>
					))}
					<div>
						<label>
							<strong>Blur Factor</strong>
						</label>
						<FloatSlider
							title="factor"
							min={0}
							max={8}
							step={0.2}
							handleChange={onChange}
							value={factor}
						/>
					</div>
					<div>
						<label><strong>Blur Passes</strong></label>
						<FloatSlider
							title="passes"
							min={1}
							max={8}
							step={1}
							handleChange={onChange}
							value={passes}
						/>
					</div>
					<div>
						<label><strong>Saturation</strong></label>
						<FloatSlider
							title="saturation"
							min={0}
							max={2}
							step={0.05}
							handleChange={onChange}
							value={saturation}
						/>
					</div>
					<div>
						<label><strong>Brightness</strong></label>
						<FloatSlider
							title="brightness"
							min={0}
							max={2}
							step={0.05}
							handleChange={onChange}
							value={brightness}
						/>
					</div>
					<div>
						<label><strong>Contrast</strong></label>
						<FloatSlider
							title="contrast"
							min={0}
							max={2}
							step={0.05}
							handleChange={onChange}
							value={contrast}
						/>
					</div>

					<StaticBlurMap handleChange={onStaticBlurMapChange} />

				</div>
			</form>
		</div>
	);
};

VideoControls.propTypes = {
	color: PropTypes.string.isRequired,
	factor: PropTypes.number.isRequired,
	passes: PropTypes.number.isRequired,
	contrast: PropTypes.number.isRequired,
	saturation: PropTypes.number.isRequired,
	brightness: PropTypes.number.isRequired,
	onStaticBlurMapChange: PropTypes.func.isRequired,
	handleColorChange: PropTypes.func.isRequired,
	handleClick: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
};

VideoControls.defaultProps = {};

export default VideoControls;
