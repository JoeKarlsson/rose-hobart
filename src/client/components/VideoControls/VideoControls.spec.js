import React from 'react';
import renderer from 'react-test-renderer';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import VideoControls from './VideoControls';

configure({ adapter: new Adapter() });

describe('VideoControls', () => {
	let wrapper;
	let inst;
	let mockOnStaticBlurMapChange;
	let mockHandleColorChange;
	let mockOnHandleClick;
	let mockOnChange;

	beforeEach(() => {
		mockOnStaticBlurMapChange = jest.fn();
		mockHandleColorChange = jest.fn();
		mockOnHandleClick = jest.fn();
		mockOnChange = jest.fn();

		wrapper = shallow(
			<VideoControls
				color="monochrome"
				factor={1}
				passes={1}
				contrast={1}
				saturation={1}
				brightness={1}
				onStaticBlurMapChange={mockOnStaticBlurMapChange}
				handleColorChange={mockHandleColorChange}
				handleClick={mockOnHandleClick}
				onChange={mockOnChange}
			/>,
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});
			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					factor: 1,
					passes: 1,
					contrast: 1,
					saturation: 1,
					brightness: 1,
					onStaticBlurMapChange: mockOnStaticBlurMapChange,
					handleClick: mockOnHandleClick,
					onChange: mockOnChange,
				};
				expect(initialProps).toMatchObject(expectedProps);
			});
			it('should not have any inital state', () => {
				const initialState = inst.state;
				const expectedState = {};
				expect(initialState).toMatchObject(expectedState);
			});
		});
	});
});
