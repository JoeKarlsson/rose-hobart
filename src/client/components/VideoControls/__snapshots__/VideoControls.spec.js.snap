// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoControls rendering initial state match the snapshot 1`] = `
<div
  className="VideoControls row"
>
  <form
    className="col s12"
  >
    <div
      className="row"
    >
      <button
        className="waves-effect waves-light btn"
        onClick={[Function]}
      >
        <i
          className="material-icons left"
        >
          play_circle_outline
        </i>
      </button>
      <button
        className="waves-effect waves-light btn"
        onClick={[Function]}
      >
        <i
          className="material-icons left"
        >
          pause_circle_outline
        </i>
      </button>
      <button
        className="waves-effect waves-light btn"
        onClick={[Function]}
      >
        <i
          className="material-icons left"
        >
          volume_off
        </i>
      </button>
      <div>
        <label>
          <strong>
            Blur Factor
          </strong>
        </label>
        <input
          max={8}
          min={0}
          onChange={[Function]}
          step={0.2}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
          type="range"
          value={0}
        />
      </div>
      <div>
        <label>
          <strong>
            Blur Passes
          </strong>
        </label>
        <input
          max={8}
          min={1}
          onChange={[Function]}
          step={1}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
          type="range"
          value={0}
        />
      </div>
      <div>
        <label>
          <strong>
            Saturation
          </strong>
        </label>
        <input
          max={2}
          min={0}
          onChange={[Function]}
          step={0.05}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
          type="range"
          value={0}
        />
      </div>
      <div>
        <label>
          <strong>
            Brightness
          </strong>
        </label>
        <input
          max={2}
          min={0}
          onChange={[Function]}
          step={0.05}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
          type="range"
          value={0}
        />
      </div>
      <div>
        <label>
          <strong>
            Contrast
          </strong>
        </label>
        <input
          max={2}
          min={0}
          onChange={[Function]}
          step={0.05}
          style={
            Object {
              "flex": 1,
              "width": "100%",
            }
          }
          type="range"
          value={0}
        />
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
            "justifyContent": "space-around",
          }
        }
      >
        <button
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
        >
          <img
            alt="Blur"
            src="https://i.imgur.com/SzbbUvX.png"
            style={
              Object {
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderWidth": 2,
                "width": 80,
              }
            }
          />
        </button>
        <button
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
        >
          <img
            alt="Blur"
            src="https://i.imgur.com/0PkQEk1.png"
            style={
              Object {
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderWidth": 2,
                "width": 80,
              }
            }
          />
        </button>
        <button
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
        >
          <img
            alt="Blur"
            src="https://i.imgur.com/z2CQHpg.png"
            style={
              Object {
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderWidth": 2,
                "width": 80,
              }
            }
          />
        </button>
        <button
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
        >
          <img
            alt="Blur"
            src="https://i.imgur.com/k9Eview.png"
            style={
              Object {
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderWidth": 2,
                "width": 80,
              }
            }
          />
        </button>
        <button
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
        >
          <img
            alt="Blur"
            src="https://i.imgur.com/wh0On3P.png"
            style={
              Object {
                "borderColor": "transparent",
                "borderStyle": "solid",
                "borderWidth": 2,
                "width": 80,
              }
            }
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
